src/common/stringutils.cpp

```diff
@@ -168,30 +223,146 @@ std::wstring StrUtils::Utf8StringToUnicode(const std::string &str)
     return result;
 }
 
-int StrUtils::Utf8CharSizeAt(const std::string &str, unsigned int pos)
+/**
+ * @brief      get the nb of char to encode current utf8 element
+ *
+ * @param[in]  str   input string
+ * @param[in]  pos   position into string
+ * @param[in]  size  preknown size of str (avoid new strlen)
+ *
+ * @return     nb of char equivalent for current displayed utf8 character
+ */
+unsigned short StrUtils::Utf8CharSizeAt(const char*const str, const std::size_t pos, const std::size_t size)
 {
-    if (pos >= str.size())
+    //  memo UTF8:
```

Comment:
I recognize this function, gave me some headache when I was debugging a few crashing issues. This is the point where I must say I am too afraid to merge this PR.